File: test-suite/src/test/java/io/micronaut/kafka/docs/quickstart/ProductListener.java
package io.micronaut.kafka.docs.quickstart;

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaKey;
import io.micronaut.configuration.kafka.annotation.KafkaListener;
import io.micronaut.configuration.kafka.annotation.OffsetReset;
import io.micronaut.configuration.kafka.annotation.Topic;
import io.micronaut.context.annotation.Requires;
import org.slf4j.Logger;

import static org.slf4j.LoggerFactory.getLogger;
// end::imports[]

@Requires(property = "spec.name", value = "QuickstartTest")
// tag::clazz[]
@KafkaListener(offsetReset = OffsetReset.EARLIEST) // <1>
public class ProductListener {
    private static final Logger LOG = getLogger(ProductListener.class);

    @Topic("my-products") // <2>
    public void receive(@KafkaKey String brand, String name) { // <3>
        LOG.info("Got Product - {} by {}", name, brand);
    }
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/quickstart/ProductListener.py
import logging
from typing import Annotated

# tag::imports[]
from micronaut.configuration.kafka.annotation import KafkaKey, KafkaListener, OffsetReset, Topic
from micronaut.context.annotation import Requires
# end::imports[]

LOG = logging.getLogger(__name__)


@Requires(property="spec.name", value="QuickstartTest")
# tag::clazz[]
@KafkaListener(offsetReset=OffsetReset.EARLIEST)  # <1>
class ProductListener:

    @Topic("my-products")  # <2>
    def receive(self, brand: Annotated[str, KafkaKey], name: str) -> None:  # <3>
        LOG.info("Got Product - %s by %s", name, brand)
# end::clazz[]
